presto-geospatial/src/main/java/com/facebook/presto/geospatial/BingTile.java

```diff
@@ -34,6 +38,39 @@ private BingTile(int x, int y, int zoomLevel)
         this.zoomLevel = zoomLevel;
     }
 
+    @Override
+    public boolean equals(Object other)
+    {
+        if (this == other) {
+            return true;
+        }
+
+        if (other == null || getClass() != other.getClass()) {
+            return false;
+        }
+
+        BingTile otherTile = (BingTile) other;
+        return this.x == otherTile.x
+            && this.y == otherTile.y
+            && this.zoomLevel == otherTile.zoomLevel;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(x, y, zoomLevel);
+    }
+
+    @Override
+    public String toString()
+    {
+        return toStringHelper(this)
+            .add("x", x)
+            .add("y", y)
+            .add("zoom_level", zoomLevel)
+            .toString();
+    }
+
     public static BingTile fromCoordinates(int x, int y, int zoomLevel)
```

Comment:
You should mark this method as `@JsonCreator` and the parameters with `@JsonProperty` so that jackson can create instances of this bean.